lua/ttt2/libraries/game_effects.lua

```diff
@@ -148,6 +148,75 @@ function gameEffects.RadiusDamage(dmginfo, pos, radius, inflictor)
     end
 end
 
+-- Creates explosion damage in a sphere through walls. Very useful for making explosives that aren't line of sight based.
+-- @param Player dmgowner The player that causes this explosion.
+-- @param Entity source The entity that causes this explosion.
+-- @param number damage The maximum damage done by this explosion.
+-- @param Vector origin The center of the explosion.
+-- @param number outerRadius The outer border for the explosion damage and its falloff.
+-- @param number innerRadius The inner border for the explosion damage where falloff starts.
+-- @param boolean exponentialFalloff If the damage falloff should be FALSE: Linear, TRUE: exponential.
+-- @internal
+-- @realm server
+function gameEffects.ExplosiveSphereDamage(
+    dmgowner,
+    source,
+    damage,
+    origin,
+    outerRadius,
+    innerRadius,
+    exponentialFalloff
+)
+    -- It seems intuitive to use FindInSphere here, but that will find all ents
+    -- in the radius, whereas there exist only ~16 players. Hence it is more
+    -- efficient to cycle through all those players and do a Lua-side distance
+    -- check.
+
+    if outerRadius < innerRadius then
+        ErrorNoHalt(
+            "[Game Effects Explosive Sphere Damage] Outer radius too high! Setting both radi to outer radius."
+        )
+        innerRadius = outerRadius
+    end
+
+    -- pre-declare to avoid realloc
+    local d = 0.0
+    local dFraction = 0.0
+    local diff = nil
+    local dmg = 0
+    local radiDiff = (outerRadius - innerRadius)
+    local plys = player.GetAll()
+    for i = 1, #plys do
+        local ply = plys[i]
+        if IsValid(ply) and ply:Team() == TEAM_TERROR then
+            diff = origin - ply:GetPos()
```

Comment:
```suggestion
            diff = origin - (ply:GetPos() + ply:OBBCenter())
```

While this doesn't _really_ matter, this takes the center of the player. I used this for all distance calculations in the code base and it doesn't cost much.

Reply:
Should i also use this for SetDamagePosition then?

Reply:
In theory, yes. However the C4 is so small I did not consider this at all. But yes, please do!